homeassistant/components/splunk/__init__.py

```diff
@@ -55,45 +56,55 @@
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the Splunk component."""
-    conf = config[DOMAIN]
-    host = conf.get(CONF_HOST)
-    port = conf.get(CONF_PORT)
-    token = conf.get(CONF_TOKEN)
-    use_ssl = conf[CONF_SSL]
-    verify_ssl = conf.get(CONF_VERIFY_SSL)
-    name = conf.get(CONF_NAME)
-    entity_filter = conf[CONF_FILTER]
-
-    event_collector = hass_splunk(
+    """Old setup for Splunk component."""
+    if DOMAIN in config:
+        # Entity filters cannot be stored in Config Flow, so must be removed
```

Comment:
We need to store the config in `hass.data` so we can apply the filter from `async_setup_entry`.